app/src/main/java/com/github/se/eventradar/model/repository/event/MockEventRepository.kt

```diff
@@ -81,4 +82,64 @@ class MockEventRepository : IEventRepository {
       }
     }
   }
+
+  override suspend fun addAttendee(eventId: String, attendeeUserId: String): Resource<Unit> {
+    return when (val res = getEvent(eventId)) {
+      is Resource.Success -> {
+        res.data?.attendeeList?.add(attendeeUserId)
+        Resource.Success(Unit)
+      }
+      is Resource.Failure -> {
+        Resource.Failure(res.throwable)
+      }
+    }
+  }
+
+  override suspend fun removeAttendee(eventId: String, attendeeUserId: String): Resource<Unit> {
+    return when (val res = getEvent(eventId)) {
+      is Resource.Success -> {
+        res.data?.attendeeList?.remove(attendeeUserId)
+        Resource.Success(Unit)
+      }
+      is Resource.Failure -> {
+        Resource.Failure(res.throwable)
+      }
+    }
+  }
+
+  override suspend fun incrementPurchases(eventId: String): Resource<Unit> {
+    return when (val res = getEvent(eventId)) {
+      is Resource.Success -> {
+        val ticket = res.data?.ticket
+        if (ticket != null) {
+          if (ticket.purchases < ticket.capacity) {
+            res.data.ticket =
+                EventTicket(ticket.name, ticket.price, ticket.capacity, ticket.purchases + 1)
+          } else {
+            return Resource.Failure(Exception("No more ticket"))
+          }
+        }
+        Resource.Success(Unit)
+      }
+      is Resource.Failure -> {
+        Resource.Failure(res.throwable)
+      }
+    }
+  }
+
+  override suspend fun decrementPurchases(eventId: String): Resource<Unit> {
+    return when (val res = getEvent(eventId)) {
+      is Resource.Success -> {
+        val ticket = res.data?.ticket
+        if (ticket != null) {
```

Comment:
nit: maybe include a check here too if purchases is equal to zero, so that it cannot be decremented anymore

Reply:
sure ! ty !